src/map/skill.c

```diff
@@ -20612,6 +20612,50 @@ static bool skill_parse_row_changematerialdb(char *split[], int columns, int cur
 	return true;
 }
 
+/**
+ * Enable Bard/Dancer to learn songs from their counterpart when spirit link
+ */
+static void skill_bard_dancer_song_soullink(struct map_session_data *sd)
+{
+	nullpo_retv(sd);
+	
+	if (sd->sc.count == 0 || sd->sc.data[SC_SOULLINK] == NULL || sd->sc.data[SC_SOULLINK]->val2 != SL_BARDDANCER)
+		return;
+	
+	int bard_song_skillid[4] = {BA_WHISTLE, BA_ASSASSINCROSS, BA_POEMBRAGI, BA_APPLEIDUN};
+	int dancer_song_skillid[4] = {DC_HUMMING, DC_DONTFORGETME, DC_FORTUNEKISS, DC_SERVICEFORYOU};
+	
+	STATIC_ASSERT(ARRAYLENGTH(bard_song_skillid) == ARRAYLENGTH(dancer_song_skillid), "bard_song_skillid and dancer_song_skillid must be the same size");
+
+	int copy_from_index;
+	int copy_to_index;
+	for (int i = 0; i < 4; i++) {
+		copy_from_index = 0;
+		copy_to_index = 0;
```

Comment:
why are you setting it to 0 when you'll always overwrite it after that? This is unnecessary.